### ISSUE TITLE
Migrate `signup.sourcegraph.com` from Netlify to GCP

### ISSUE THREAD

**Issue by sanderginn** [sourcegraph/sourcegraph#42815]
Compatibility between Netlify Functions and Cloud SQL is suboptimal. ~Migrating to Cloud Functions will solve the pain points we observed.~

Instead of migrating to Cloud Functions, the entire application will be Dockerized and deployed to the kubernetes cluster of dotcom.

https://github.com/sourcegraph/sourcegraph/issues/42815#tasklist-block-f59428a7-837e-4f42-a9a6-14dcba53bc7c

### Nice to haves
- [ ] Deployment previews à la Netlify (until then Netlify can still be leveraged for this)

**Comment by rafax** [comment]
Is there more context why we decided to reuse DotCom cluster instead of the alternatives (Cloud Run being what seems like a natural fit, with a clear path to go back to K8S if needed)?

**Comment by sanderginn** [comment]
@rafax I decided for the dotcom cluster due to
1. it being a fairly simple deployment that would be easy to schedule on cluster resources that otherwise would be idle
2. it being a `.sourcegraph.com` domain (which to be fair is not a particularly strong argument given that we have plenty of other workloads using similar subdomains that are managed separately)

I don't have a stronger argument than this, so I'm happy to adopt Cloud Run instead if that has your preference.

**Comment by sanderginn** [comment]
@rafax there's an amount of additional work required to move this to the dotcom cluster that makes Cloud Run definitely the easier approach, so that's what we'll be doing. Thanks for the input!

**Comment by rafax** [comment]
Understood, as you mentioned colocating with DotCom requires additional work, and Cloud Run is a perfect fit for "stateless Container listening on a single port"

**Comment by sanderginn** [comment]
Closing as the work is done, just needs validating:
- Prod works as intended (will be done by @erzhtor)
- DNS entry is switched over to Cloud Run when ready
- Netlify will run in parallel as a fallback for a while. When it's shut down, merge https://github.com/sourcegraph/infrastructure/pull/4457

**Comment by erzhtor** [comment]
@sanderginn, I'll reopen it until we switch signup.sourcegraph.com to cloud run prod env, if you don't mind. 
- This will help to create sub-issues that will be part of this one
- Also, some other issues, such https://github.com/sourcegraph/signup/issues/316 cannot be done, until we don't switch to Cloud Run, so I need to "blocked by" issue to clarify things

**Comment by sanderginn** [comment]
@erzhtor ack 👍

**Comment by sanderginn** [comment]
Issue no longer relevant.
